_aggregations/metric/scripted-metric.md

```diff
@@ -9,44 +9,220 @@ redirect_from:
-* `combine_script`: Aggregates the state returned from every shard. The aggregated value is returned to the coordinating node.
-* `reduce_script`: Provides access to the variable states; this variable combines the results from the `combine_script` on each shard into an array.
+## Parameters
 
-The following example aggregates the different HTTP response types in web log data:
+The `scripted_metric` aggregation takes the following parameters:
+
+| Parameter        | Data type | Required/Optional | Description                                                                                        |
+| ---------------- | --------- | ----------------- | -------------------------------------------------------------------------------------------------- |
+| `init_script`    | String    | Optional          | The script executes once per shard before any documents are processed. Used to set up an initial state. For example, initialize counters or lists in a state object. If not provided, the state starts as an empty object on each shard.       |
+| `map_script`     | String    | Required          | The script executes for each document collected by the aggregation. This script updates the state based on the document’s data. For example, you might check the field's value and then increment a counter or sum a running total in the state.                                                  |
+| `combine_script` | String    | Required          | The script executes once per shard after all documents on that shard have been processed by the `map_script`. This script aggregates the shard’s state into a single result to be sent back to the coordinating node. This script is used to finalize the computation for one shard. For example, summing up counters or totals stored in state. The script should return the consolidated value or structure for its shard.  |
+| `reduce_script`  | String    | Required          | The script executes once on the coordinating node after receiving combined results from all shards. This script receives a special variable states, which is an array containing each shard’s output from the combine script. The `reduce_script` iterates over states and produces the final aggregation output. For example, summing shard sums, or merging maps of counts. The value returned by the reduce script will be the value reported in the aggregation results. |
+| `params`         | Object    | Optional          | User-defined parameters accessible from all scripts except `reduce_script`.                        |
+
+## Allowed return types
+
+Scripts can use any valid operations and objects internally, however the data you store in `state`, or return from any script, must be of permitted types. This restriction exists because the intermediate state needs to be sent between nodes. The allowed types are:
+
+- Primitive types: `int`, `long`, `float`, `double`, `boolean`
+- String
+- Map (with keys and values only of allowed types)
```

Comment:
`Map (with keys and values only of allowed types)` : What allowed types? String & Primitives defined above?